src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.c

```diff
@@ -28,61 +46,391 @@ ActorInit En_Takaraya_InitVars = {
+    if (!sTexturesDesegmented) {
+        for (i = 0; i < ARRAY_COUNT(sEyesDownTextures); i++) {
+            sEyesUpTextures[i] = Lib_SegmentedToVirtual(sEyesUpTextures[i]);
+            sEyesDownTextures[i] = Lib_SegmentedToVirtual(sEyesDownTextures[i]);
+        }
+        sTexturesDesegmented = true;
+    }
+    this->eyeTexIndex = 2;
+    if (gSaveContext.save.entrance == ENTRANCE(TREASURE_CHEST_SHOP, 1)) {
+        Audio_StopSubBgm();
+        if (gSaveContext.timerStates[TIMER_ID_MINIGAME_2] == TIMER_STATE_6) {
+            this->actor.textId = 0x77A;
+            gSaveContext.timerStates[TIMER_ID_MINIGAME_2] = TIMER_STATE_STOP;
+            func_80ADF6DC(this);
+        } else {
+            this->actor.textId = 0x77C;
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED)) {
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED);
+                EnTakaraya_SetupWait(this);
+            } else {
+                func_80ADF6DC(this);
+            }
+        }
+    } else {
+        EnTakaraya_SetupWait(this);
+    }
+}
+
+void EnTakaraya_Destroy(Actor* thisx, PlayState* play) {
+    EnTakaraya* this = THIS;
+
+    Flags_UnsetSwitch(play, 5);
+    if (!this->unk2AD) {
+        CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+        gSaveContext.timerStates[TIMER_ID_MINIGAME_2] = TIMER_STATE_STOP;
+    }
+}
+
+void EnTakaraya_Blink(EnTakaraya* this) {
+    if (this->eyeTexIndex != 0) {
+        this->eyeTexIndex++;
+        if (this->eyeTexIndex == 4) {
+            this->eyeTexIndex = 0;
+        }
+    } else if (Rand_ZeroOne() < 0.02f) {
+        this->eyeTexIndex++;
+    }
+}
+
+void EnTakaraya_SetupWait(EnTakaraya* this) {
+    if (this->skelAnime.animation == &object_bg_Anim_001384) {
+        Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00AD98, 5.0f);
+    }
+    this->actionFunc = EnTakaraya_Wait;
+}
+
+void EnTakaraya_Wait(EnTakaraya* this, PlayState* play) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        if (this->skelAnime.animation == &object_bg_Anim_00A280) {
+            Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00AD98, 5.0f);
+        } else {
+            Animation_MorphToLoop(&this->skelAnime, &object_bg_Anim_009890, -4.0f);
+        }
+    }
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        if (Text_GetFaceReaction(play, FACE_REACTION_SET_TREASURE_CHEST_SHOP_GAL) == 0) {
+            Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00A280, -4.0f);
+        }
+        EnTakaraya_SetupTalk(this);
+    } else if (Actor_IsFacingPlayer(&this->actor, 0x2000)) {
+        this->actor.textId = Text_GetFaceReaction(play, FACE_REACTION_SET_TREASURE_CHEST_SHOP_GAL);
+        if (this->actor.textId == 0) {
+            this->actor.textId = D_80ADFB2C[GET_PLAYER_FORM];
+        }
+        this->formSwitchFlag = gSaveContext.save.playerForm + this->switchFlag;
```

Comment:
```suggestion
        this->formSwitchFlag = GET_PLAYER_FORM + this->switchFlag;
```
Does this still match? Same for the other 
`this->formSwitchFlag = gSaveContext.save.playerForm + this->switchFlag;`
below